### PR TITLE
Add possibility to build without musl

### LINKED ISSUE #19803: Commit a9f48bd should be reverted, the musl shouldn't be a default libc
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

This commit a9f48bd  break compilation script for glibc distors (most?! except Alpine) and where Rust managed by pkg manager (because you have `rustup` call). 

### Environment

Gentoo Linux but it doesn't matter here, it's should affect many other

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

### PR DESCRIPTION
**Pull request by stalkerg**
Closes #19803  (it's already closed, but we supposed this PR can be a good option)

In #a9f48bd, added a musl libc target for the remote server and dependencies for Debian-like distros in `scripts/Linux.` 
This breaks the bundle-linux script for non-Debian-like distros (it requires musl-gcc package). Also, it introduced `rustup` as a mandatory tool and made it difficult for people who manage rust by system managers. 

This PR provides an option to build a zed remote server with a any host target (in my case glibc):
```bash
REMOTE_SERVER_TARGET="x86_64-unknown-linux-gnu" ./script/install-linux
```

Release Notes:

- N/A

### CODE REVIEW

**Comment by stalkerg** [comment]
@cla-bot check

File: script/bundle-linux [hunk 1]
```diff
@@ -38,27 +38,31 @@ version_info=$(rustc --version --verbose)
...
 target_triple=${host_line#*: }
 musl_triple=${target_triple%-gnu}-musl
+remote_server_triple=${REMOTE_SERVER_TARGET:-"${musl_triple}"}
 
 # Generate the licenses first, so they can be baked into the binaries
 script/generate-licenses
-rustup target add "$musl_triple"
+
+if [[ "$remote_server_triple" == "$musl_triple" ]]; then
+    rustup target add "$musl_triple"
```
Comment (SomeoneToIgnore) [inline script/bundle-linux:47]: Thinking of the generic approach to this, does it make sense to install `musl_triple` at all?
Seems that we'd better add `remote_server_triple` unconditionally?

**Review by SomeoneToIgnore — COMMENTED** [review]
Looks simple enough to incorporate, thank you.

Let's make the toolchain installation a bit better and merge this?

File: script/bundle-linux (hunk 1 above)
Comment (stalkerg) [reply script/bundle-linux:47]: Probably better here, doing next: try to add target by rustup only if remote_server_triple different from host rust compiler (default target). 
If $remote_server_triple the same what `rustc --version --verbose` return is have no sens to change anything.

Comment (SomeoneToIgnore) [reply script/bundle-linux:47]: Yeah, that also could work.

My current concern is that anything can be put into `REMOTE_SERVER_TARGET` "argument", and if that is not installed along the way, we'll get more cryptic messages.

Not insisting strongly, lmk if you want to keep things as is and merge.

**Comment by SomeoneToIgnore** [comment]
I've incorporated my feedback and the feedback from https://github.com/zed-industries/zed/pull/19571#issuecomment-2450287201 and ready to merge in ~24 hours if the CI passes.

Does this work on your system? Anything you'd like to change?

**Comment by stalkerg** [comment]
@SomeoneToIgnore sorry for delay! Yes, I think it's looks good to me. Thank you.

**Pull request merged**